File: 1-100/53.maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if (nums.empty())
        {
            return 0;
        }

        int curSum = nums[0];
        int maxSum = curSum;
        size_t nSize = nums.size();
        for (size_t i = 1; i < nSize; ++i)
        {
            curSum = curSum > 0 ? curSum + nums[i] : nums[i];
            maxSum = maxSum > curSum ? maxSum : curSum;
        }

        return maxSum;
    }
};

File: 1-100/53.maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if not nums:
            return 0
        
        curSum = nums[0]
        maxSum = curSum
        for num in nums[1:]:
            curSum = curSum + num if curSum > 0 else num
            maxSum = maxSum if maxSum > curSum else curSum
        
        return maxSum
